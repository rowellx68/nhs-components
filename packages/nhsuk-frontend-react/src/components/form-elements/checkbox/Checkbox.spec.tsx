import React from 'react';
import { it, expect } from 'vitest';
import { render } from '@testing-library/react';
import { composeStory } from '@storybook/react';
import meta, {
  Default as DefaultStory,
  WithHint as WithHintStory,
  WithError as WithErrorStory,
  WithItemHint as WithItemHintStory,
  WithConditionalContent as WithConditionalContentStory,
  WithExclusiveOption as WithExclusiveOptionStory,
} from './Checkbox.stories';

const Default = composeStory(DefaultStory, meta);
const WithHint = composeStory(WithHintStory, meta);
const WithError = composeStory(WithErrorStory, meta);
const WithItemHint = composeStory(WithItemHintStory, meta);
const WithConditionalContent = composeStory(WithConditionalContentStory, meta);
const WithExclusiveOption = composeStory(WithExclusiveOptionStory, meta);

it('should render the Checkbox component', () => {
  const { container } = render(<Default />);

  expect(container.querySelector('input[type="checkbox"]')).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});

it('should render the Checkbox with exclusive option', () => {
  const { container } = render(<WithExclusiveOption />);

  expect(container.querySelector('input[type="checkbox"]')).toBeInTheDocument();
  expect(
    container.querySelector('.nhsuk-checkboxes__divider'),
  ).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});

it('should render the Checkbox with conditional content', () => {
  const { container } = render(<WithConditionalContent />);

  expect(container.querySelector('input[type="email"]')).toBeInTheDocument();
  expect(container.querySelector('input[type="tel"]')).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});

it('should render the Checkbox with hint', () => {
  const { container } = render(<WithHint />);

  expect(container.querySelector('.nhsuk-hint')).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});

it('should render the Checkbox with item hint', () => {
  const { container } = render(<WithItemHint />);

  expect(container.querySelector('.nhsuk-hint')).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});

it('should render the Checkbox with error', () => {
  const { container } = render(<WithError />);

  expect(container.querySelector('.nhsuk-error-message')).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});
