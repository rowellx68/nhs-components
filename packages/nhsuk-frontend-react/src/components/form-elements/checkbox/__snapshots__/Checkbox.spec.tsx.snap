// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render the Checkbox component 1`] = `
<div>
  <form>
    <div
      class="nhsuk-form-group"
    >
      <fieldset
        aria-labelledby="checkboxes_:r0:--label"
        class="nhsuk-fieldset"
      >
        <legend
          class="nhsuk-legend nhsuk-fieldset__legend--l"
          for="checkboxes_:r0:"
          id="checkboxes_:r0:--label"
        >
          <h1
            class="nhsuk-fieldset__heading"
          >
            How would you like to be contacted?
          </h1>
        </legend>
        <div
          as="div"
          class="nhsuk-checkboxes"
          id="checkboxes_:r0:"
        >
          <div
            class="nhsuk-checkboxes__item"
          >
            <input
              class="nhsuk-checkboxes__input"
              id="checkbox-item_:r2:"
              type="checkbox"
              value="email"
            />
            <label
              class="nhsuk-label nhsuk-checkboxes__label"
              for="checkbox-item_:r2:"
            >
              Email
            </label>
          </div>
          <div
            class="nhsuk-checkboxes__item"
          >
            <input
              class="nhsuk-checkboxes__input"
              id="checkbox-item_:r5:"
              type="checkbox"
              value="phone"
            />
            <label
              class="nhsuk-label nhsuk-checkboxes__label"
              for="checkbox-item_:r5:"
            >
              Phone
            </label>
          </div>
          <div
            class="nhsuk-checkboxes__item"
          >
            <input
              class="nhsuk-checkboxes__input"
              id="checkbox-item_:r8:"
              type="checkbox"
              value="sms"
            />
            <label
              class="nhsuk-label nhsuk-checkboxes__label"
              for="checkbox-item_:r8:"
            >
              Text message
            </label>
          </div>
        </div>
      </fieldset>
    </div>
  </form>
</div>
`;

exports[`should render the Checkbox with conditional content 1`] = `
<div>
  <form>
    <div
      class="nhsuk-form-group"
    >
      <div
        class="nhsuk-form-group"
      >
        <fieldset
          aria-describedby="checkboxes_:rp:_:rq:--hint"
          aria-labelledby="checkboxes_:rp:--label"
          class="nhsuk-fieldset"
        >
          <legend
            class="nhsuk-legend nhsuk-fieldset__legend--l"
            for="checkboxes_:rp:"
            id="checkboxes_:rp:--label"
          >
            <h1
              class="nhsuk-fieldset__heading"
            >
              How would you like to be contacted?
            </h1>
          </legend>
          <div
            class="nhsuk-hint"
            id="checkboxes_:rp:_:rq:--hint"
          >
            Select all options that are relevant to you.
          </div>
          <div
            as="div"
            class="nhsuk-checkboxes nhsuk-checkboxes--conditional"
            id="checkboxes_:rp:"
          >
            <div
              class="nhsuk-checkboxes__item"
            >
              <input
                aria-controls="checkbox-item-conditional_:r1c:"
                aria-expanded="false"
                class="nhsuk-checkboxes__input"
                id="checkbox-item_:r1a:"
                type="checkbox"
                value="email"
              />
              <label
                class="nhsuk-label nhsuk-checkboxes__label"
                for="checkbox-item_:r1a:"
              >
                Email
              </label>
            </div>
            <div
              class="nhsuk-checkboxes__conditional nhsuk-checkboxes__conditional--hidden"
              id="checkbox-item-conditional_:r1c:"
            >
              <div
                class="nhsuk-form-group"
              >
                <label
                  class="nhsuk-label"
                  for="input_:r1d:"
                  id="input_:r1d:--label"
                >
                  Email address
                </label>
                <input
                  aria-labelledby="input_:r1d:--label"
                  as="input"
                  class="nhsuk-input nhsuk-u-width-two-thirds"
                  id="input_:r1d:"
                  type="email"
                />
              </div>
            </div>
            <div
              class="nhsuk-checkboxes__item"
            >
              <input
                aria-controls="checkbox-item-conditional_:r1h:"
                aria-expanded="false"
                class="nhsuk-checkboxes__input"
                id="checkbox-item_:r1f:"
                type="checkbox"
                value="phone"
              />
              <label
                class="nhsuk-label nhsuk-checkboxes__label"
                for="checkbox-item_:r1f:"
              >
                Phone
              </label>
            </div>
            <div
              class="nhsuk-checkboxes__conditional nhsuk-checkboxes__conditional--hidden"
              id="checkbox-item-conditional_:r1h:"
            >
              <div
                class="nhsuk-form-group"
              >
                <label
                  class="nhsuk-label"
                  for="input_:r1i:"
                  id="input_:r1i:--label"
                >
                  Phone number
                </label>
                <input
                  aria-labelledby="input_:r1i:--label"
                  as="input"
                  class="nhsuk-input nhsuk-u-width-two-thirds"
                  id="input_:r1i:"
                  type="tel"
                />
              </div>
            </div>
            <div
              class="nhsuk-checkboxes__item"
            >
              <input
                aria-controls="checkbox-item-conditional_:r1m:"
                aria-expanded="false"
                class="nhsuk-checkboxes__input"
                id="checkbox-item_:r1k:"
                type="checkbox"
                value="sms"
              />
              <label
                class="nhsuk-label nhsuk-checkboxes__label"
                for="checkbox-item_:r1k:"
              >
                Text message
              </label>
            </div>
            <div
              class="nhsuk-checkboxes__conditional nhsuk-checkboxes__conditional--hidden"
              id="checkbox-item-conditional_:r1m:"
            >
              <div
                class="nhsuk-form-group"
              >
                <label
                  class="nhsuk-label"
                  for="input_:r1n:"
                  id="input_:r1n:--label"
                >
                  Mobile number
                </label>
                <input
                  aria-labelledby="input_:r1n:--label"
                  as="input"
                  class="nhsuk-input nhsuk-u-width-two-thirds"
                  id="input_:r1n:"
                />
              </div>
            </div>
          </div>
        </fieldset>
      </div>
    </div>
  </form>
</div>
`;

exports[`should render the Checkbox with error 1`] = `
<div>
  <form>
    <div
      class="nhsuk-form-group nhsuk-form-group--error"
    >
      <fieldset
        aria-describedby="checkboxes_:r2f:--error-message"
        aria-labelledby="checkboxes_:r2f:--label"
        class="nhsuk-fieldset"
      >
        <legend
          class="nhsuk-legend nhsuk-fieldset__legend--l"
          for="checkboxes_:r2f:"
          id="checkboxes_:r2f:--label"
        >
          <h1
            class="nhsuk-fieldset__heading"
          >
            How would you like to be contacted?
          </h1>
        </legend>
        <span
          class="nhsuk-error-message"
          id="checkboxes_:r2f:--error-message"
        >
          <span
            class="nhsuk-u-visually-hidden"
          >
            Error: 
          </span>
          Select one option
        </span>
        <div
          as="div"
          class="nhsuk-checkboxes"
          id="checkboxes_:r2f:"
        >
          <div
            class="nhsuk-checkboxes__item"
          >
            <input
              class="nhsuk-checkboxes__input"
              id="checkbox-item_:r2h:"
              type="checkbox"
              value="email"
            />
            <label
              class="nhsuk-label nhsuk-checkboxes__label"
              for="checkbox-item_:r2h:"
            >
              Email
            </label>
          </div>
          <div
            class="nhsuk-checkboxes__item"
          >
            <input
              class="nhsuk-checkboxes__input"
              id="checkbox-item_:r2k:"
              type="checkbox"
              value="phone"
            />
            <label
              class="nhsuk-label nhsuk-checkboxes__label"
              for="checkbox-item_:r2k:"
            >
              Phone
            </label>
          </div>
          <div
            class="nhsuk-checkboxes__item"
          >
            <input
              class="nhsuk-checkboxes__input"
              id="checkbox-item_:r2n:"
              type="checkbox"
              value="sms"
            />
            <label
              class="nhsuk-label nhsuk-checkboxes__label"
              for="checkbox-item_:r2n:"
            >
              Text message
            </label>
          </div>
        </div>
      </fieldset>
    </div>
  </form>
</div>
`;

exports[`should render the Checkbox with exclusive option 1`] = `
<div>
  <form>
    <div
      class="nhsuk-form-group"
    >
      <fieldset
        aria-describedby="checkboxes_:rb:_:rc:--hint"
        aria-labelledby="checkboxes_:rb:--label"
        class="nhsuk-fieldset"
      >
        <legend
          class="nhsuk-legend nhsuk-fieldset__legend--l"
          for="checkboxes_:rb:"
          id="checkboxes_:rb:--label"
        >
          <h1
            class="nhsuk-fieldset__heading"
          >
            Do you have any of these symptoms?
          </h1>
        </legend>
        <div
          class="nhsuk-hint"
          id="checkboxes_:rb:_:rc:--hint"
        >
          Select all the symptoms you have.
        </div>
        <div
          as="div"
          class="nhsuk-checkboxes"
          id="checkboxes_:rb:"
        >
          <div
            class="nhsuk-checkboxes__item"
          >
            <input
              class="nhsuk-checkboxes__input"
              data-checkbox-exclusive-group="symptoms-list"
              id="checkbox-item_:rd:"
              type="checkbox"
              value="sore throat"
            />
            <label
              class="nhsuk-label nhsuk-checkboxes__label"
              for="checkbox-item_:rd:"
            >
              Sore throat
            </label>
          </div>
          <div
            class="nhsuk-checkboxes__item"
          >
            <input
              class="nhsuk-checkboxes__input"
              data-checkbox-exclusive-group="symptoms-list"
              id="checkbox-item_:rg:"
              type="checkbox"
              value="runny nose"
            />
            <label
              class="nhsuk-label nhsuk-checkboxes__label"
              for="checkbox-item_:rg:"
            >
              Runny nose
            </label>
          </div>
          <div
            class="nhsuk-checkboxes__item"
          >
            <input
              class="nhsuk-checkboxes__input"
              data-checkbox-exclusive-group="symptoms-list"
              id="checkbox-item_:rj:"
              type="checkbox"
              value="muscle or joint pain"
            />
            <label
              class="nhsuk-label nhsuk-checkboxes__label"
              for="checkbox-item_:rj:"
            >
              Muscle or joint pain
            </label>
          </div>
          <div
            class="nhsuk-checkboxes__divider"
          >
            or
          </div>
          <div
            class="nhsuk-checkboxes__item"
          >
            <input
              class="nhsuk-checkboxes__input"
              data-checkbox-exclusive="true"
              data-checkbox-exclusive-group="symptoms-list"
              id="checkbox-item_:rm:"
              type="checkbox"
              value="none"
            />
            <label
              class="nhsuk-label nhsuk-checkboxes__label"
              for="checkbox-item_:rm:"
            >
              No, I do not have any of these symptoms
            </label>
          </div>
        </div>
      </fieldset>
    </div>
  </form>
</div>
`;

exports[`should render the Checkbox with hint 1`] = `
<div>
  <form>
    <div
      class="nhsuk-form-group"
    >
      <fieldset
        aria-describedby="checkboxes_:r1p:_:r1q:--hint"
        aria-labelledby="checkboxes_:r1p:--label"
        class="nhsuk-fieldset"
      >
        <legend
          class="nhsuk-legend nhsuk-fieldset__legend--l"
          for="checkboxes_:r1p:"
          id="checkboxes_:r1p:--label"
        >
          <h1
            class="nhsuk-fieldset__heading"
          >
            How would you like to be contacted?
          </h1>
        </legend>
        <div
          class="nhsuk-hint"
          id="checkboxes_:r1p:_:r1q:--hint"
        >
          Select all options that are relevant to you.
        </div>
        <div
          as="div"
          class="nhsuk-checkboxes"
          id="checkboxes_:r1p:"
        >
          <div
            class="nhsuk-checkboxes__item"
          >
            <input
              class="nhsuk-checkboxes__input"
              id="checkbox-item_:r1r:"
              type="checkbox"
              value="email"
            />
            <label
              class="nhsuk-label nhsuk-checkboxes__label"
              for="checkbox-item_:r1r:"
            >
              Email
            </label>
          </div>
          <div
            class="nhsuk-checkboxes__item"
          >
            <input
              class="nhsuk-checkboxes__input"
              id="checkbox-item_:r1u:"
              type="checkbox"
              value="phone"
            />
            <label
              class="nhsuk-label nhsuk-checkboxes__label"
              for="checkbox-item_:r1u:"
            >
              Phone
            </label>
          </div>
          <div
            class="nhsuk-checkboxes__item"
          >
            <input
              class="nhsuk-checkboxes__input"
              id="checkbox-item_:r21:"
              type="checkbox"
              value="sms"
            />
            <label
              class="nhsuk-label nhsuk-checkboxes__label"
              for="checkbox-item_:r21:"
            >
              Text message
            </label>
          </div>
        </div>
      </fieldset>
    </div>
  </form>
</div>
`;

exports[`should render the Checkbox with item hint 1`] = `
<div>
  <form>
    <div
      class="nhsuk-form-group"
    >
      <fieldset
        aria-describedby="checkboxes_:r24:_:r25:--hint"
        aria-labelledby="checkboxes_:r24:--label"
        class="nhsuk-fieldset"
      >
        <legend
          class="nhsuk-legend nhsuk-fieldset__legend--l"
          for="checkboxes_:r24:"
          id="checkboxes_:r24:--label"
        >
          <h1
            class="nhsuk-fieldset__heading"
          >
            What is your nationality?
          </h1>
        </legend>
        <div
          class="nhsuk-hint"
          id="checkboxes_:r24:_:r25:--hint"
        >
          If you have dual nationality, select all options that are relevant to you.
        </div>
        <div
          as="div"
          class="nhsuk-checkboxes"
          id="checkboxes_:r24:"
        >
          <div
            class="nhsuk-checkboxes__item"
          >
            <input
              aria-describedby="checkbox-item-hint_:r27:"
              class="nhsuk-checkboxes__input"
              id="checkbox-item_:r26:"
              type="checkbox"
              value="email"
            />
            <label
              class="nhsuk-label nhsuk-checkboxes__label"
              for="checkbox-item_:r26:"
            >
              British
            </label>
            <div
              class="nhsuk-hint nhsuk-checkboxes__hint"
              id="checkbox-item-hint_:r27:"
            >
              including English, Scottish, Welsh and Northern Irish
            </div>
          </div>
          <div
            class="nhsuk-checkboxes__item"
          >
            <input
              class="nhsuk-checkboxes__input"
              id="checkbox-item_:r29:"
              type="checkbox"
              value="phone"
            />
            <label
              class="nhsuk-label nhsuk-checkboxes__label"
              for="checkbox-item_:r29:"
            >
              Irish
            </label>
          </div>
          <div
            class="nhsuk-checkboxes__item"
          >
            <input
              class="nhsuk-checkboxes__input"
              id="checkbox-item_:r2c:"
              type="checkbox"
              value="sms"
            />
            <label
              class="nhsuk-label nhsuk-checkboxes__label"
              for="checkbox-item_:r2c:"
            >
              Citizen of another country
            </label>
          </div>
        </div>
      </fieldset>
    </div>
  </form>
</div>
`;
